src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AutoValue
+public abstract class JavaDocComment implements Comment {
+
+  public abstract ImmutableList<String> comments();
+
+  public abstract ImmutableList<String> params();
+
+  public abstract String deprecated();
+
+  // Private.
+  abstract String throwType();
```

Comment:
If we keep this, let's name it `throes` or some similar misspelled variant, so we don't have to use `throwType`.

Reply:
so here `throwType` only gets the type of `@throws`, it also has `throwDescription` 
> * @throws com.google.api.gax.rpc.ApiException.     if the remote call fails.

we do not have a getter for throws, when users call `setThrows(throwType, throwDescription)`, `throwType` and `throwDescription` will be set separately. And `comment()` will call their private getters to construct the consolidated comment.

Reply:
Thanks for clarifying. I think we don't need these getter methods (as they will never be used), please consider removing them.

Reply:
as the example mentioned above `name` in MethodDefinition class, setter in Builder class needs a getter as a field, private or public. Tried to remove the getter the got the below error:
```
error: Method does not correspond to a property of com.google.api.generator.engine.ast.JavaDocComment
```
Please LMK if I'm understanding wrong, thanks!

Reply:
Please see the comment above.